packages/web-components/src/components/va-banner/va-banner.tsx

```diff
@@ -0,0 +1,185 @@
+   * */
+  @Prop() type: string = 'info';
+
+  /**
+   * A boolean that when false makes it so that the banner does not render.
+   * */
+  @Prop() visible: boolean = true;
+
+  /**
+   * Enable sessionStorage for the Banner otherwise storage
+   * if showClose is enabled will default to localStorage
+   * */
+  @Prop() windowSession: boolean = false;
+
+  /**
+   * Keep track of locally dismissed Banners
+   * */
+  @State() dismissedBanners: Array<String> = [];
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link or the dismiss icon is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private prepareBannerID = () => `${this.headline}:${this.el.innerHTML}`;
+
+  private dismiss = () => {
+    // Derive the current banner ID.
+    const currentBannerID = this.prepareBannerID();
+
+    // Escape early if the banner is already dismissed.
+    if (this.dismissedBanners.includes(currentBannerID)) {
+      return;
+    }
+
+    // Add the banner ID to the list of dismissed banners. If showClose enabled
+    if (this.showClose) {
+      // Derive the updated dismissed banners.
+      const updatedDismissedBanners = [
+        ...this.dismissedBanners,
+        currentBannerID,
+      ];
+
+      //   Set storage type
+      const storageType = this.windowSession
+        ? window.sessionStorage
+        : window.localStorage;
+
+      // Set banner dismissed in storage
+      storageType.setItem(
+        DISMISSED_BANNERS_KEY,
+        JSON.stringify(updatedDismissedBanners),
+      );
+
+      // Update dismissedBanners in state.
+      this.dismissedBanners = updatedDismissedBanners;
+
+      // Track the dismiss event in Google Analytics
+      if (!this.disableAnalytics) {
+        const detail = {
+          componentName: 'Banner',
+          action: 'dismissIconClick',
+          details: {
+            clickLabel: 'Dismiss Banner',
+            headline: this.headline,
+            showClose: this.showClose,
+            windowSession: this.windowSession,
+            type: this.type,
+          },
+        };
+        this.componentLibraryAnalytics.emit(detail);
+      }
+    }
+  };
+
+  private handleAlertBodyClick(e: MouseEvent): void {
+    if (!this.disableAnalytics) {
+      const target = e.target as HTMLElement;
+      // If it's a link being clicked, dispatch an analytics event
+      if (target?.tagName === 'A') {
+        const detail = {
+          componentName: 'Banner',
+          action: 'linkClick',
+          details: {
+            clickLabel: target.innerText,
+            headline: this.headline,
+            showClose: this.showClose,
+            type: this.type,
+          },
+        };
+        this.componentLibraryAnalytics.emit(detail);
+      }
+    }
+  }
```

Comment:
@bkjohnson good point on the double analytics event firing. I had added it in because in the React Component version  of the Banner we had a recordEvent() being fired when an A tag was clicked in the banner here: https://github.com/department-of-veterans-affairs/component-library/blob/b44be8aa94e7833604ef367b5a6481fdd8f0d2e5/packages/react-components/src/components/Banner/Banner.jsx#L109

But since we are already tracking that event in the Alert Web component I guess we probably don't need it in the banner. Do you think we should have different events for the A tag clicking in the different components or is this a larger question that we need to discuss with the Analytics team?